autopilot-daemon/network/iperf3_entrypoint.py

```diff
@@ -63,16 +63,28 @@ async def make_server_connection(event, address, handle):
 
 async def make_client_connection(event, iface, src, dst, address, handle):
     # Task waits for the event to be set before starting its work.
-    if event != None:
-        await event.wait()
-    url = f"http://{address}:{AUTOPILOT_PORT}{handle}"
-    total_timeout = aiohttp.ClientTimeout(total=60 * 10)
-    async with aiohttp.ClientSession(timeout=total_timeout) as session:
-        async with session.get(url) as resp:
-            reply = await resp.text()
-            reply = reply.strip()
-            json_reply = json.loads(reply)
-            return {"src": src, "dst": dst, "iface": iface, "data": json_reply}
+    try:
+        if event != None:
+            await event.wait()
+        url = f"http://{address}:{AUTOPILOT_PORT}{handle}"
+        total_timeout = aiohttp.ClientTimeout(total=60 * 10)
+        async with aiohttp.ClientSession(timeout=total_timeout) as session:
+            async with session.get(url) as resp:
+                reply = await resp.text()
+                reply = "".join(reply.split())
+                try:
+                    json_reply = json.loads(reply)
+                except json.JSONDecodeError as e:
+                    log.error(
+                        f"Failed to decode JSON from response: {e}. Response: {reply}"
+                    )
+                    return {"src": src, "dst": dst, "iface": iface, "data": {}}
+
+                return {"src": src, "dst": dst, "iface": iface, "data": json_reply}
+    except Exception as e:
+        log.error(f"Error during client connection to {address} at {handle}: {e}")
+        log.error(f"Failure occured with from src {src} to dst {dst} on iface {iface}")
+        return {"src": src, "dst": dst, "iface": iface, "data": {}}
```

Comment:
Don't crash -- let the test continue to aggregate all failures. Rather than stop prematurely.